hook.sh

```diff
@@ -46,17 +46,40 @@ echo -e "\e[32mSetting challenge to ${CERTBOT_VALIDATION} ...\e[0m"
 domain=.$CERTBOT_DOMAIN
 infix=${domain%.$DEDYN_NAME}
 
+# Remove leading wildcard from infix, if present
+# *.foobar.dedyn.io gives "" while *.a.foobar.dedyn.io gives ".a"
+infix=${infix#.\*}
+
 args=( \
-    '-Ss' \
+    '-sSLf' \
     '-H' "Authorization: Token $DEDYN_TOKEN" \
     '-H' 'Accept: application/json' \
     '-H' 'Content-Type: application/json' \
-    '-d' '[{"subname":"_acme-challenge'"$infix"'", "type":"TXT", "records":["\"'"$CERTBOT_VALIDATION"'\""], "ttl":60}]' \
-    '-o' '/dev/null' \
 )
 
+# Letsencrypt uses a TTL of 60 seconds on their resolvers. If we're getting
+# wildcard certificates (e.g. foobar.dedyn.io and *.foobar.dedyn.io), multiple
+# certificates for the same domain (e.g. RSA and ECDSA), or both, we'll need to
+# wait 60 seconds for the records to expire before verification begins. We'll
+# store any existing challenges and publish them along with the current one.
+# Set timeout_ttl=yes in ./dedynauth if you want to enable this.
+# You'll probably also want to use the optional cleanup hook to keep records
+# from accumulating.
+
+readarray -t acme_live <<< "$(curl "${args[@]}" -X GET "https://desec.io/api/v1/domains/$DEDYN_NAME/rrsets/_acme-challenge$infix/TXT" \
+    | grep -o '"records":\[[^]]*]' | grep -o '\\"[^,]*\\"')"
+
+[ ${#acme_live[@]} -gt 0 ] && [ -n "${acme_live[*]}" ] && [ -n "$timeout_ttl" ] && \
+echo -e "\e[32mWaiting 60 seconds for Letsencrypt resolvers to clear ...\e[0m" &&  sleep 60
+
+for ((i = 0; i < "${#acme_live[@]}"; i++));do
```

Comment:
If we don't split the TXT records further up, we don't need the loop here. We can just check whether `current_value='"\"'"$CERTBOT_VALIDATION"'\""'` is a substring of the records string, and if it's not the case, add `,${current_value}` to the string.

Reply:
As above :-)